From f793f3f364cb0e5ba210c8d9d50b284395cde3a9 Mon Sep 17 00:00:00 2001
From: Vincenzo Pupillo <v.pupillo@gmail.com>
Date: Wed, 26 Aug 2026 21:15:49 +0200
Subject: [PATCH] 'speedbar-window' now run 'speedbar-before-delete-hook'
 (bug#81708).

* lisp/speedbar.el (speedbar-window-mode): Fix documentation. Run the
'speedbar-before-delete-hook'.
---
 lisp/speedbar.el | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lisp/speedbar.el b/lisp/speedbar.el
index 198a309065d..3f904d9c6db 100644
--- a/lisp/speedbar.el
+++ b/lisp/speedbar.el
@@ -1064,10 +1064,10 @@ speedbar-window-mode
 A nil ARG means toggle.  Once the speedbar window is activated, a buffer in
 `speedbar-mode' will be displayed.  Currently, only one speedbar is
 supported at a time.
-`speedbar-before-popup-hook' is called before popping up the speedbar frame.
 `speedbar-before-delete-hook' is called before the frame is deleted."
   (interactive "P")
   (when (eq (speedbar-frame-or-window) 'frame)
+    (run-hooks 'speedbar-before-delete-hook)
     (delete-frame (speedbar-current-frame)))
 
   (if (or (and (not arg) (speedbar-window--live-p))
-- 
2.55.0

